AgoraSocialApp/Areas/Identity/Pages/Account/Login.cshtml.cs

```diff
@@ -55,21 +53,12 @@ public class InputModel
             public bool RememberMe { get; set; }
         }
 
-        public async Task OnGetAsync(string returnUrl = null)
+        public IActionResult OnGetAsync(string returnUrl = null)
         {
-            if (!string.IsNullOrEmpty(ErrorMessage))
-            {
-                ModelState.AddModelError(string.Empty, ErrorMessage);
-            }
-
-            returnUrl ??= Url.Content("~/");
-
-            // Clear the existing external cookie to ensure a clean login process
-            await HttpContext.SignOutAsync(IdentityConstants.ExternalScheme);
-
-            ExternalLogins = (await _signInManager.GetExternalAuthenticationSchemesAsync()).ToList();
-
-            ReturnUrl = returnUrl;
+            var provider = "Google";
+            var redirectUrl = Url.Page("./ExternalLogin", pageHandler: "Callback", values: new { returnUrl });
+            var properties = _signInManager.ConfigureExternalAuthenticationProperties(provider, redirectUrl);
+            return new ChallengeResult(provider, properties);
```

Comment:
We're going to very quickly add other identity providers... let's not lock ourselves in to Google and lose some of this potentially useful code

Reply:
agree, i added the acr_values to the oidc client already but need to add the di so i can read the value here

Reply:
figured out how to read the acr_values in .net core and now have the sveltekit app set up to send values to .net who can then route to the correct providers